soc/arm/st_stm32/stm32h7/CMakeLists.txt

```diff
@@ -4,6 +4,7 @@ zephyr_include_directories(${ZEPHYR_BASE}/drivers)
 
 zephyr_sources_ifdef(CONFIG_CPU_CORTEX_M7 soc_m7.c)
 zephyr_sources_ifdef(CONFIG_CPU_CORTEX_M7 power_m7.c)
+zephyr_sources_ifdef(CONFIG_SOC_STM32H735XX power_stm32h735xx.c)
```

Comment:
Are there some reasons it can't be more widely used ?

Reply:
So far I based my work upon the reference manual for H723, H733, H725, H735 and H730. Therefore I will have to check if the same procedures also apply to H743, H745, H747, H750 and H753.
As soon as I did so I will update this line.

Reply:
Thanks.
It's possible that there are some differences between these SoC.
Maybe we'll need to introduce some device tree configuration to be able to group them by specifics

Reply:
I went through the other reference manuals, and of course there is a difference. H743, H745, H747, H750 and H753 require an additional step for changing into VOS0. For these chips this switch is only allowed from VOS1 and with the ODEN bit in SYSCFG_PWRCR set. Hence, only the voltage scaling is different.

My idea for this would be to make a separate file for the voltage scaling which gets included to the build by a new config param:

`zephyr_sources_ifdef(CONFIG_SOC_STM32H7_VOS_ODEN power_voltage_overdrive.c)
zephyr_sources_ifdef(CONFIG_SOC_STM32H7_VOS_WITHOUT_ODEN power_voltage.c)`

I would set these flags automatically based upon the chip selection.

Does this sound reasonable to you?

Reply:
I switched to a slightly different way using a menu choice. Hopefully more elegant.